src/index/CompressedRelation.cpp

```diff
@@ -164,7 +168,7 @@ CompressedRelationReader::asyncParallelBlockGenerator(
   }
   // The `OnDestruction...` above might be called too late, so we manually set
   // the timer again.
-  details.blockingTimeMs_ = popTimer.msecs();
+  storeTime();
```

Comment:
I think we either have to stop the timer before this call (then we can get rid of the whole block), or we have to cancel the Cleanup. Otherwise we probably overwrite the runtime info again with a possibly "too large" value.
This is a bug of mine.
Maybe just replace this line by `popTimer.stop()` and get rid of the comment and the explicit storeTime.

Reply:
Right, didn't consider the case where the timer wasn't stopped. In all other cases it would just redundantly store the same value twice.